great_expectations/cli/datasource.py

```diff
@@ -1282,12 +1282,9 @@ def _get_batch_kwargs_for_sqlalchemy_datasource(
             data_asset_name = "custom_sql_query"
 
         elif single_or_multiple_data_asset_selection == "3":  # list it all
-            msg_prompt_warning = fr"""Warning: If you have a large number of tables in your datasource, this may take a very long time. \m
-                    Would you like to continue?"""
-            confirmation = click.prompt(
-                msg_prompt_warning, type=click.Choice(["y", "n"]), show_choices=True
-            )
-            if confirmation == "y":
+            msg_prompt_warning = f"""Warning: If you have a large number of tables in your datasource, this may take a very long time.\nWould you like to proceed?"""
+            confirmation = click.confirm(msg_prompt_warning, default=True)
```

Comment:
This is what is used in `confirm_proceed_or_exit()` method that is used by commands in the CLI that start a potentially lengthy (>1 sec) computation or modifies some resources (e.g., edits the config file, adds objects to the stores).  The method allows the user to confirm the command or exits the program.

In the case of selecting a datasource, we don't actually want to exit GE, so we just keep the same prompt.

Reply:
Does this print the default [Y/n]?

Reply:
yes. `click.confirm()` is what prints the `[Y/n]` part. The `msg_prompt_warning` is everything leading up to it including `Would you like to proceed?`

Reply:
Nice!